### PR TITLE
Xfreerdp cleanup

### PR DESCRIPTION
**Pull request by akallabeth**
* Allow `xfreerdp` to run without `DISPLAY` variable for `/auth-only`
* Adds support for transparent windows (`RAILS`)
* Adds a `PostFinalDisconnect` client callback that is called after everything shut down (`PostDisconnect` is called before channel shutdown, this one is similar to `PreConnect`)
* Some `X11` related code cleanups, specifically drawing path and modifier map updates
* RAIL transparency in this PR currently only works with `AVC444`, other codecs still need some follow up bug fixing

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test this again please

**Comment by akallabeth** [comment]
@freerdp-bot test

File: channels/rdp2tcp/client/rdp2tcp_main.c [hunk 1]
```diff
@@ -168,7 +171,9 @@ static DWORD WINAPI copyThread(void* data)
 			goto fail;
 		}
 
-		WaitForSingleObject(plugin->writeComplete, INFINITE);
+		handles[0] = plugin->writeComplete;
+		handles[1] = freerdp_abort_event(plugin->channelEntryPoints.context);
+		status = WaitForMultipleObjects(2, handles, FALSE, INFINITE);
 		ResetEvent(plugin->writeComplete);
```
Comment (hardening) [inline channels/rdp2tcp/client/rdp2tcp_main.c:177]: Here we would ResetEvent `plugin->writeComplete` unconditionnally even if status is an error, that seems fishy

File: client/X11/xf_rail.c [hunk 2]
```diff
@@ -209,63 +210,61 @@ void xf_rail_end_local_move(xfContext* xfc, xfAppWindow* appWindow)
...
-	xf_rail_invalidate_region(xfc, &invalidRegion);
-	region16_uninit(&invalidRegion);
+	BOOL rc = TRUE;
+	ULONG_PTR* pKeys = NULL;
+	size_t count = 0;
+
+	WINPR_ASSERT(xfc);
+	WINPR_ASSERT(rect);
+
+	if (xfc->railWindows)
```
Comment (hardening) [inline client/X11/xf_rail.c:256]: Perhaps we could use `HashTable_Foreach` ?

**Review by hardening — APPROVED** [review]
A few remarks but globally LGTM

File: channels/rdp2tcp/client/rdp2tcp_main.c (hunk 1 above)
Comment (akallabeth) [reply channels/rdp2tcp/client/rdp2tcp_main.c:177]: @hardening well, as the other condition is `terminate the session` I didn´t think the check would be useful, but yes, could be added.

File: client/X11/xf_rail.c (hunk 2 above)
Comment (akallabeth) [reply client/X11/xf_rail.c]: done

File: channels/rdp2tcp/client/rdp2tcp_main.c (hunk 1 above)
Comment (akallabeth) [reply channels/rdp2tcp/client/rdp2tcp_main.c]: but fixed nevertheless.

**Pull request merged**